node/state.re

```diff
@@ -117,8 +124,46 @@ let try_to_commit_state_hash = (~old_state, state, block, signatures) => {
     );
   });
 };
-let apply_block = (state, block) => {
+let apply_block = (state, block: Block.t) => {
   let old_state = state;
+
+  let {
+    trusted_validator_membership_change,
+    protocol: {last_seen_membership_change_timestamp, _},
+    _,
+  } = state;
+  let current_time = Unix.time();
+  let next_allowed_membership_change_timestamp =
+    last_seen_membership_change_timestamp +. 24. *. 60. *. 60.;
+  let contains_only_trusted_add_validator_op =
+    Operation.(
+      List.for_all(h => {
+        switch (h.Side_chain.kind) {
+        | Add_validator(validator) =>
+          Trusted_validators_membership_change.(
+            Set.mem(
+              {address: validator.address, action: Add},
+              trusted_validator_membership_change,
+            )
+          )
+          && current_time > next_allowed_membership_change_timestamp
+        | Remove_validator(validator) =>
+          Trusted_validators_membership_change.(
+            Set.mem(
+              {address: validator.address, action: Remove},
+              trusted_validator_membership_change,
+            )
+          )
+          && current_time > next_allowed_membership_change_timestamp
+        | _ => true
+        }
+      })
+    );
+
+  let.assert () = (
+    `Invalid_block_when_applying,
+    contains_only_trusted_add_validator_op(block.side_chain_ops),
+  );
```

Comment:
This entire checking should not be here, this is a property of a signable block, you're not allowed to sign a block that contains a trusted validator change that you don't know, but you should still apply the block if 2/3 of the validators sign it.

Reply:
Fair point! How about moving this to `is_signable` ? Attempted this in the next incoming commit.